interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
+Researched answer: Parameters are the variables that are being used inside of a definition of a function. Arguments are the actual values that are being passed through the function when it is called. 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The built in map method takes three parameters; value, index, and array. I believe the first parameter value is always used, while index and array can be optional.
 
-Researched answer:
+Researched answer: The higher-order function .map() takes three parameters. Value is the only required parameter which is the value of the array. Index shows the index of the array and is optional. Array is also optional and is the array that the higher order function was used on. 
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: Map is used to return an array of the same length, after using it to modify the array. Filter is used if you want to return an array with specific parts of the array, not needing to be the same length as you are filtering out what you would want.
 
-Researched answer:
+Researched answer: Map is used to create a new array with with the same length, after being manipulated however desired. If you have an array with 5 elements, it will return 5 elements, but those 5 can be changed however you set the function to do so. Filter will create an array that can be a different length than the original array, as it only brings forward the elements that your function has been set to bring forward. If you have an array with 5 elements, but you want to filter 3 elements that share a common characteristic, you would use .filter to extract those elements, and your new array would only output those 3 elements. 
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is used in for loops to repeat a functionality repeatedly.
 
-Researched answer:
+Researched answer: Iteration is used when you want to loop a functionality repeatedly, in a predetermined amount of loops. If you wanted something to loop five times, you would use a for loop to iterate a function to loop 5 times
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class is a blueprint for objects, so objects would be within the actual class
 
-Researched answer:
+Researched answer: Class is a blueprint for objects to be created in, while Objects are just an instance inside of a class. Classes are used to group similar objects together using the 'class' keyword in PascalCase. 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: I have never heard of hoisting, but I feel like it has to do with moving some sort of functionality around in JavaScript, hoisting or pulling it somewhere within your code.
 
-Researched answer:
+Researched answer: Variable hoisting in JavaScript allows you to access variables before they're declared. 
```

Comment:
Well done overall, Nate!  I encourage you to use examples in all of your interview  answers as it shows you have a good understanding of the topic.  This will be very helpful in interviews!